clash-prelude/src/Clash/Sized/Internal/Unsigned.hs

```diff
@@ -259,14 +260,16 @@ instance KnownNat n => Bounded (Unsigned n) where
   minBound = minBound#
   maxBound = maxBound#
 
-{-# NOINLINE minBound# #-}
 minBound# :: Unsigned n
 minBound# = U 0
+{-# NOINLINE minBound# #-}
 
+maxBound# :: forall n. KnownNat n => Unsigned n
+maxBound# =
+  case natToNum @n of
+    0 -> 0
+    n -> let m = 1 `shiftL` n in  U (m - 1)
```

Comment:
Why the case? isn't `shiftL 1 0` equal to `1`

Reply:
Woops, forgot to revert that after having implemented `errorX ....`